builtin/providers/ibmcloud/provider.go

```diff
@@ -0,0 +1,81 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"ibmid": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The IBM ID.",
+				DefaultFunc: schema.EnvDefaultFunc("IBMID", ""),
+			},
+			"ibmid_password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The password for the IBM ID.",
+				DefaultFunc: schema.EnvDefaultFunc("IBMID_PASSWORD", ""),
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The Bluemix Region (for example 'ng').",
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"BM_REGION", "BLUEMIX_REGION"}, "ng"),
+			},
+			"softlayer_timeout": {
```

Comment:
With the prefix we are just making sure that it is for the softlayer API. We will add bluemix API and might have a timeout for that set differently